### PR TITLE
chef/storage: process all metrics from the sack in one single batch

### PR DESCRIPTION
**Pull request by jd**
This allows in this first step to retrieve and store the unaggregated
timeseries all at once.

### CODE REVIEW

**Comment by jd** [comment]
Fix Gabbi

File: gnocchi/storage/__init__.py [hunk 1]
```diff
@@ -487,120 +487,129 @@ def _delete_metric_splits(self, metric, keys_and_aggregations, version=3):
...
-            }
-
-            deleted_keys, keys_and_split_to_store = (
-                self._compute_split_operations(
-                    metric, aggregations_and_timeseries,
-                    current_first_block_timestamp,
-                    new_first_block_timestamp)
-            )
+        for metric, measures in six.iteritems(metrics_and_measures):
+            if len(measures) == 0:
```
[the remark this answers is not in the document]
Comment (jd) [reply gnocchi/storage/__init__.py]: I've removed this entirely. It dates back from < 2.0 (see commit d8170f43af936bc1f56f5f18488294ca5858d7c1) where we did not have sack locking and worker could be stepping on each other toes. This is not true anymore, and no empty measures list will ever be passed to this function.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1f361bd4b1b0d6b62c31a581892d05b9a2637702

File: gnocchi/storage/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chungg) [reply gnocchi/storage/__init__.py]: even better

**Pull request merged**